openmls/src/framing/plaintext.rs

```diff
@@ -228,6 +229,29 @@ impl MlsPlaintext {
         )
     }
 
+    /// This constructor builds an `MlsPlaintext` containing an External Proposal.
+    /// The sender type is either `Sender::Preconfigured` or `Sender::NewMember`.
+    pub(crate) fn member_external_proposal(
```

Comment:
I'd make this exclusively about `NewMember` for now because this API doesn't force the caller to choose the right sender type and that's unsafe.